Better file naming for AppVeyor artifacts

Ashesh3 opened the pull request:
This builds upon https://github.com/godotengine/godot/pull/38625 and improves the file name format as mentioned in https://github.com/godotengine/godot/pull/38189#issuecomment-626585363

The resulting file stored has the name `godot.windows.opt.tools.64-$(VERSION_HASH).zip` where `VERSION_HASH` is obtained from `it rev-parse --short=9 HEAD` , the zip file contains the built exe file.

akien-mga commented:
IMO it would be good to also change the name of the .exe, so that users can remember what extracted binary corresponds to.

Might also be good to add the `$env:APPVEYOR_REPO_BRANCH` (that's in PS, would have to check how to get it in cmd or convert the script to `ps:`) which could be `3.2` or `master`.

E.g. some naming like: `godot_${APPVEYOR_REPO_BRANCH}-${COMMIT}_win64.exe`.

BTW there's also `APPVEYOR_REPO_COMMIT` in the env that gives the commit hash, though I don't know if it's the full version or a shortened one.

Ashesh3 commented:
> IMO it would be good to also change the name of the .exe, so that users can remember what extracted binary corresponds to.
> 
> Might also be good to add the `$env:APPVEYOR_REPO_BRANCH` (that's in PS, would have to check how to get it in cmd or convert the script to `ps:`) which could be `3.2` or `master`.
> 
> E.g. some naming like: `godot_${APPVEYOR_REPO_BRANCH}-${COMMIT}_win64.exe`.
> 
> BTW there's also `APPVEYOR_REPO_COMMIT` in the env that gives the commit hash, though I don't know if it's the full version or a shortened one.
> 
> https://www.appveyor.com/docs/environment-variables/

COMMIT is the full hash , i suggest we still use `git rev-parse --short=9 HEAD` for that, but `APPVEYOR_REPO_BRANCH` works fine, 

Here's the filenames after my updates PR:
`godot_3.2-6abcde465_win64.exe` 
inside
`godot_3.2-6abcde465_win64.zip`

akien-mga commented on .appveyor.yml:
```diff
@@ -41,9 +41,13 @@ build_script:
-  - path: godot_build.zip
-    name: Build
+  - path: bin/godot_${APPVEYOR_REPO_BRANCH}-${VERSION_HASH}_win64.zip
+    name: release_debug
```
```suggestion
    name: Win64 release_debug editor build
```

akien-mga replied:
Could you also squash commits into one?

akien-mga commented on .appveyor.yml:
```diff
@@ -41,9 +41,13 @@ build_script:
 
 after_build:
-  - 7z a godot_build.zip bin\*.exe
+  - git rev-parse --short=9 HEAD>VERSION_HASH.txt
+  - set /P VERSION_HASH=<VERSION_HASH.txt
```
```suggestion
  - git rev-parse --short=9 HEAD > VERSION_HASH.txt
  - set /P VERSION_HASH= < VERSION_HASH.txt
```
(Provided that this works on Windows - if not, keep as is.)

akien-mga commented:
Could you squash commits into one?
See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase).

Ashesh3 commented:
I think i did squash them into one commit - https://github.com/Ashesh3/godot/commit/18d71fad6b022802ca1877db1659f19e323f8dc3

Is that correct?

akien-mga commented:
> Is that correct?

Not exactly, as you also included 6 unrelated commits that you rebased (that's partly due to my rebase instructions in the doc being not optimal). You should rebase with `git rebase -i upstream/master` to fix it.

Ashesh3 committed:
Better file naming for AppVeyor artifacts

This builds upon https://github.com/godotengine/godot/pull/38625 and improves the file name format as mentioned in https://github.com/godotengine/godot/pull/38189#issuecomment-626585363

The resulting file stored has the name `godot.windows.opt.tools.64-$(VERSION_HASH).zip` where `VERSION_HASH` is obtained from `it rev-parse --short=9 HEAD` , the zip file contains the built exe file.

Ashesh3 commented:
Oh i might have done something work, anyways, it should be fine now. 

Thanks @Technohacker for helping me making this work.

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 3.2.2.
